spec/03-plugins/38-ai-proxy/09-streaming_integration_spec.lua

```diff
@@ -0,0 +1,514 @@
+local helpers = require "spec.helpers"
+local cjson = require "cjson.safe"
+local pl_file = require "pl.file"
+
+local http = require("resty.http")
+
+local PLUGIN_NAME = "ai-proxy"
+local MOCK_PORT = helpers.get_available_port()
+
+for _, strategy in helpers.all_strategies() do if strategy ~= "cassandra" then
+  describe(PLUGIN_NAME .. ": (access) [#" .. strategy .. "]", function()
+    local client
+
+    lazy_setup(function()
+      local bp = helpers.get_db_utils(strategy == "off" and "postgres" or strategy, nil, { PLUGIN_NAME })
+
+      -- set up openai mock fixtures
+      local fixtures = {
+        http_mock = {},
+        dns_mock = helpers.dns_mock.new({
+          mocks_only = true,      -- don't fallback to "real" DNS
+        }),
+      }
+
+      fixtures.dns_mock:A {
+        name = "api.openai.com",
+        address = "127.0.0.1",
+      }
+
+      fixtures.dns_mock:A {
+        name = "api.cohere.com",
+        address = "127.0.0.1",
+      }
+
+      fixtures.http_mock.streams = [[
+        server {
+          server_name openai;
+          listen ]]..MOCK_PORT..[[;
+
+          default_type 'application/json';
+          chunked_transfer_encoding on;
+          proxy_buffering on;
+          proxy_buffer_size 600;
+          proxy_buffers 10 600;
+
+          location = "/openai/llm/v1/chat/good" {
+            content_by_lua_block {
+              local _EVENT_CHUNKS = {
```

Comment:
protip: if you don't want to maintain this whole blob of code as a string, you could make it a fixture instead.

1. stick all the relevant code somewhere in a module in `spec/fixtures/...`
2. replace this with `require("spec.fixtures.my-fixture-name").some_request_handler()`

Reply:
Can I park this one? Again there's gonna be a giant re-write of this all soon, which will include real integration testing with all supported LLM.

Reply:
Yeah absolutely!